lib/client/api.go

```diff
@@ -1857,14 +1888,21 @@ func (tc *TeleportClient) getProxySSHPrincipal() string {
 	return proxyPrincipal
 }
 
-// authMethods returns a list (slice) of all SSH auth methods this client
-// can use to try to authenticate
-func (tc *TeleportClient) authMethods() []ssh.AuthMethod {
-	m := append([]ssh.AuthMethod(nil), tc.Config.AuthMethods...)
+// authMethods returns a slice of all SSH auth methods this client
+// can use to try to authenticate.
+func (tc *TeleportClient) authMethods() ([]ssh.AuthMethod, error) {
+	m := append([]ssh.AuthMethod{}, tc.Config.AuthMethods...)
 	if tc.localAgent != nil {
-		m = append(m, tc.localAgent.AuthMethods()...)
+		agentMethods, err := tc.localAgent.AuthMethods()
+		if len(m) == 0 && err != nil {
+			return nil, trace.Wrap(err)
+		}
+		m = append(m, agentMethods...)
+	}
+	if len(m) == 0 {
+		return nil, trace.BadParameter("no auth method available")
```

Comment:
```suggestion
		return nil, trace.NotFound("no SSH auth method available, try logging in again")
```

Reply:
I wouldn't reformulate in this way as it would make the relogin log unreadable. Current version:
```
DEBU [CLIENT]    Activating relogin on no auth method available. client/api.go:455
```
Also, `RetryWithRelogin` wouldn't handle `NotFoundError` as well:
https://github.com/gravitational/teleport/blob/879f8c2b486809298ecddea842ce4f64de2e655f/lib/client/api.go#L447-L451

Reply:
OK, as long as this error is never returned to users in non-debug output

Reply:
Or maybe it would actually help in handling the errors more correctly when the user has to re-login manually....